src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx

```diff
@@ -4,20 +4,59 @@
  */
 
 import { User, WorkPackageStage } from 'shared';
+import { useState } from 'react';
 import { fullNamePipe } from '../../../utils/pipes';
 import PageBlock from '../../../layouts/PageBlock';
 import { FormControl, FormLabel, Grid, MenuItem, TextField } from '@mui/material';
 import ReactHookTextField from '../../../components/ReactHookTextField';
 import { Controller } from 'react-hook-form';
 import { DatePicker } from '@mui/x-date-pickers';
+import NERAutocomplete from '../../../components/NERAutocomplete';
 
 interface Props {
-  users: User[];
+  usersForPL: User[];
+  usersForPM: User[];
   control: any;
   errors: any;
 }
 
-const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) => {
+const WorkPackageEditDetails: React.FC<Props> = ({ usersForPL, usersForPM, control, errors }) => {
+  const [lead, setLead] = useState<User | null>(null);
+  const [manager, setManager] = useState<User | null>(null);
+
+  const leadSearchOnChange = (_event: React.SyntheticEvent<Element, Event>, value: { label: string; id: string } | null) => {
+    if (value) {
+      const lead = usersForPL.find((lead: User) => lead.userId.toString() === value.id);
+      if (lead) {
+        setLead(lead);
+      }
+    } else {
+      setLead(null);
+    }
+  };
+
+  const managerSearchOnChange = (
+    _event: React.SyntheticEvent<Element, Event>,
+    value: { label: string; id: string } | null
+  ) => {
+    if (value) {
+      const manager = usersForPM.find((manager: User) => manager.userId.toString() === value.id);
+      if (manager) {
+        setManager(manager);
+      }
+    } else {
+      setManager(null);
+    }
+  };
+
+  const projectLeadOptions = (lead: User): { label: string; id: string } => {
```

Comment:
these two functions can be combined into one since tehy do the exact same thing -- and should be renamed to convertUserToOption so that it's clear that it's a function and stuff